### DISCUSSION TITLE
How come the Prisma typescript code is not generated using Rust?

### DISCUSSION THREAD

**Discussion by aaronvg** [prisma/prisma#21607] [Q&A] [answered]
### Question

I was wondering why the decision was made to split the generation to be done in typescript, rather than in the `engines` packages, where you already have the whole AST built up.

**Comment by ludralph** [comment]
Hi @aaronvg 👋 

Thank you for raising this question.

I have shared your question internally with the engineering team. I'll provide a response as soon as I get one.

**Reply by ludralph** [reply, marked as answer]
Hi @aaronvg 

Thank you for your patience.

Some of the reasons I could gather on why the generation was done in Typescript was because:

1. Rust/Engines deal with Schema file, Queries, Migrations etc. Typescript/CLI/Client deals with everything else. Also a big impact probably also had that the people knowing TS (and the types etc) were more comfortable to write that in TS as well.

2. Code generation is easier to do in TypeScript since the output is also JS/TS.

3. The engines were supposed to be language agnostic. Back then we thought we would have support for many languages soon. So we wanted to establish a pattern that a new client can be added without the need to touch the Rust part.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by aaronvg** [reply]
Thanks for getting this info! And thank you to the prisma team for responding :)
